tools/fork-network/src/cli.rs

```diff
@@ -0,0 +1,253 @@
+                let entry = entry?;
+                if entry.file_type()?.is_file() {
+                    std::fs::remove_file(&entry.path())?;
+                }
+            }
+            println!("Restoring fork snapshot");
+            for entry in std::fs::read_dir(&fork_snapshot_path)? {
+                let entry = entry?;
+                std::fs::rename(&entry.path(), &store_path.join(entry.file_name()))?;
+            }
+            std::fs::remove_dir(&fork_snapshot_path)?;
+
+            println!("Restoring genesis file");
+            let genesis_path = home_dir.join(near_config.config.genesis_file.clone() + ".backup");
+            let original_genesis_path = home_dir.join(&near_config.config.genesis_file);
+            std::fs::rename(&genesis_path, &original_genesis_path)?;
+            return Ok(());
+        }
+        let near_config = load_config(home_dir, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+
+        if !near_config.config.store.flat_storage_creation_enabled {
+            println!(
+                "Flat storage is disabled, so the forked network will also not have flat storage"
+            );
+        }
+
+        if near_config.validator_signer.is_none() {
+            panic!("Validator ID is required; please set up validator_keys.json");
+        }
+
+        let store_opener = NodeStorage::opener(
+            home_dir,
+            near_config.config.archive,
+            &near_config.config.store,
+            None,
+        );
+
+        let store_path =
+            home_dir.join(near_config.config.store.path.clone().unwrap_or(PathBuf::from("data")));
+        let migration_snapshot_path = store_path.join(
+            near_config
+                .config
+                .store
+                .migration_snapshot
+                .get_path(&store_path)
+                .expect("Migration snapshot must be enabled"),
+        );
+        let fork_snapshot_path = store_path.join("fork-snapshot");
+
+        println!("Creating snapshot of original db into {}", fork_snapshot_path.display());
+        store_opener.create_snapshots(Mode::ReadWrite)?;
+        std::fs::rename(&migration_snapshot_path, &fork_snapshot_path)?;
+
+        let storage = store_opener.open_in_mode(Mode::ReadWrite).unwrap();
+        let store = storage.get_hot_store();
+
+        let epoch_manager =
+            EpochManager::new_arc_handle(store.clone(), &near_config.genesis.config);
+        let runtime = NightshadeRuntime::from_config(
+            home_dir,
+            store.clone(),
+            &near_config,
+            epoch_manager.clone(),
+        );
+        let head = store.get_ser::<Tip>(DBCol::BlockMisc, HEAD_KEY)?.unwrap();
+        let shard_layout = epoch_manager.get_shard_layout(&head.epoch_id)?;
+        let all_shard_uids = shard_layout.get_shard_uids();
+
+        let fork_head = if near_config.config.store.flat_storage_creation_enabled {
```

Comment:
We can stop considering `flat_storage_creation_enabled` here and above, because flat storage creation already happened. Comparing heads still makes a lot of sense though. Once we start supporting catchups, sometimes heads can be way behind and we may not want to fork node in such state 🤔